include/operators/bitwise.hpp

```diff
@@ -0,0 +1,384 @@
+*/
+
+BigInt BigInt::operator|(const BigInt& rhs) const {
+    // Neither side can be negative
+    if (rhs < 0) {
+        throw std::invalid_argument("Operator '|' is undefined on negative right-hand argument");
+    }
+    if (*this < 0)
+    {
+        throw std::invalid_argument("Operator '|' is undefined on negative left-hand argument");
+    }
+    BigInt left = *this;
+    BigInt right = rhs;
+    BigInt result = 0; 
+    BigInt increment = 1;
+    
+    while (left > 0 || right > 0)
+    {
+        if ((left % 2 == 1) || (right % 2 == 1))    
+        {
+            result += increment; 
+        } 
+        increment *= 2; 
+        left /= 2;   
+        right /= 2;
+    }
+    return result; 
+}
+
+
+/*
+    BigInt & BigInt
+    ---------------
+    Returns a BigInt with value equal to the bitwise AND of the binary representations of two positive operands
+*/
+
+BigInt BigInt::operator&(const BigInt& rhs) const {
+    // Neither side can be negative
+    if (rhs < 0) {
+        throw std::invalid_argument("Operator '&' is undefined on negative right-hand argument");
+    }
+    if (*this < 0)
+    {
+        throw std::invalid_argument("Operator '&' is undefined on negative left-hand argument");
+    }
+    BigInt left = *this;
+    BigInt right = rhs;
+    BigInt result = 0; 
+    BigInt increment = 1;
+    
+    while (left > 0 || right > 0)
+    {
+        if ((left % 2 == 1) && (right % 2 == 1))    
+        {
+            result += increment; 
+        } 
+        increment *= 2; 
+        left /= 2;   
+        right /= 2;
+    }
+    return result;
+}
+
+
+
+/*
+    BigInt ^ BigInt
+    ---------------
+    Returns a BigInt with value equal to the bitwise XOR of the binary representations of two positive operands
+*/
+
+
+BigInt BigInt::operator^(const BigInt& rhs) const {
+    // Neither side can be negative
+    if (rhs < 0) {
+        throw std::invalid_argument("Operator '^' is undefined on negative right-hand argument");
+    }
+    if (*this < 0)
+    {
+        throw std::invalid_argument("Operator '^' is undefined on negative left-hand argument");
+    }
+    BigInt left = *this;
+    BigInt right = rhs;
+    BigInt result = 0; 
+    BigInt increment = 1;
+    
+    while (left > 0 || right > 0)
+    {
+        bool left_bit = (left % 2 == 1);
+        bool right_bit = (right % 2 == 1);
+        if ((left_bit || right_bit) && (left_bit != right_bit))    
+        {
+            result += increment; 
+        } 
+        increment *= 2; 
+        left /= 2;   
+        right /= 2;
+    }
+    return result;
+}
+
+
+
+/*
+    ~BigInt
+    ---------------
+    Returns a BigInt with value equal to the bitwise NOT of the binary representations of the positive input
+*/
+
+
+BigInt BigInt::operator~() const {
+    // Operand cannot be negative
+    if (*this < 0)
+    {
+        throw std::invalid_argument("Operator '~' is undefined on negative operand");
+    }
+    else if(*this == 0)
```

Comment:
This was a quick fix, but in reality when we want to do ~0, we need to know how many bits we are working with, since
~0x0000 = 0xFFFF or ~0x0 =0xF.
So it seems that we have an undefined behavior here. How should I proceed?
Maybe this should case should also yield an exception.